staging/src/k8s.io/apiserver/pkg/storage/cacher/caching_object.go

```diff
@@ -56,12 +56,12 @@ type serializationResult struct {
 // serializationsCache is a type for caching serialization results.
 type serializationsCache map[runtime.Identifier]*serializationResult
 
-// cachingObject is an object that is able to cache its serializations
+// CachingObject is an object that is able to cache its serializations
 // so that each of those is computed exactly once.
 //
-// cachingObject implements the metav1.Object interface (accessors for
+// CachingObject implements the metav1.Object interface (accessors for
 // all metadata fields).
-type cachingObject struct {
+type CachingObject struct {
```

Comment:
I think I would prefer avoiding making this public, but I see that it's making this benchmark fairly simple to add.

@liggitt - WDYT about making this public?

Reply:
I really don't want to make these public... this was a pretty messy implementation detail to begin with and I don't want any chance of accumulating external use

Reply:
+1 this can't be public

Reply:
I reimplemented simpler version.